src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,162 @@
+
+      if (data.isAuthenticated) {
+        setIsAuthenticated(true);
+      } else {
+        setIsAuthenticated(false);
+        getCSRF();
+      }
+
+      setLoading(false);
+    } catch (err) {
+      console.error("Failed to fetch session data:", err);
+      setIsAuthenticated(false);
+      setLoading(false);
+    }
+  };
+
+  const getCSRF = async (): Promise<void> => {
+    try {
+      const res = await api.get('/api/csrf_token/');
+
+      const csrfToken = res.headers['x-csrftoken']; 
+      if (csrfToken) {
+        setCsrf(csrfToken);
+        devPrint("CSRF Token fetched and set:", csrfToken);
+      } else {
+        throw new Error("CSRF token not found in response headers");
+      }
+    } catch (err) {
+      console.error("Failed to fetch CSRF token:", err);
+    }
+  };
+
+  const login = async (username: string, password: string): Promise<void> => {
+    try {
+      const res = await api.post('/api/user/login/', {
+        username,
+        password,
+      });
+
+      if (res.status === 200) {
+        setIsAuthenticated(true);
+        setError("");
+      } else {
+        const errorData = res.data;
+        if (errorData.detail === "Your account does not have a Discord ID associated with it.") {
+          setError(`Your discord is not verified. Please type /verify ${errorData.username} in the swecc server`);
+        } else {
+          console.error("Login failed:", errorData);
+          setError("Invalid credentials. Please try again.");
+          setIsAuthenticated(false);
+        }
+      }
+    } catch (err: any) {
+      if (err.response) {
+        console.error("Error during login:", err.response.data);
+        setError(err.response.data.detail || "Login failed. Please try again.");
+      } else {
+        console.error("Error during login:", err);
+        setError("An error occurred. Please try again later.");
+      }
+      setIsAuthenticated(false);
+    }
+  };
+
+  const logout = async (): Promise<void> => {
+    try {
+      const res = await api.post('/api/user/logout/');
+
+      if (res.status === 200) {
+        devPrint("Logout successful");
+        setIsAuthenticated(false);
+        await getCSRF();
+      } else {
+        console.error("Logout failed");
+      }
+    } catch (err) {
+      console.error("Error during logout:", err);
+    }
+  };
+
+  const register = async (username: string, password: string, discord_username: string): Promise<void> => {
+    try {
+      const res = await api.post('/api/user/register/', {
+        username,
+        password,
+        discord_username,
+      });
+
+      if (res.status !== 201) {
+        throw new Error("Registration failed.");
+      }
+
+      const data = res.data;
+      devPrint("Registration successful:", data);
+      setError("");
+      //This shouldn't be an error but too lazy right now to create redirect page
+      setError(`Registration successful. Please type /verify ${username} in the swecc server`);
```

Comment:
/auth not /verify

Reply:
This was already fixed before you left the comments 😛

Reply:
ah, dirty read